setupdune.sh

```diff
@@ -5,10 +5,31 @@
 set -euo pipefail
 
 abort() {
-  printf 'Fatal error: %s\n' "$1" 1>&2
+  printf '::error title=Fatal error::%s\n' "$1" 1>&2
   exit 2
 }
 
+dune_aux() {
+  status=0
+  (set -x; cd "$SETUPDUNEDIR" && \
+    dune "$@" \
+      ${SETUPDUNEWORKSPACE:+--workspace="$SETUPDUNEWORKSPACE"} \
+      ${SETUPDUNEDISPLAY:+--display="$SETUPDUNEDISPLAY"}) \
+    || status=$?
+  if ! test "$status" = 0; then
+    if test -e "$SETUPDUNEDIR/_build/log"; then
+      echo "::endgroup::"
+      echo '::group::`_build/log`'
+      printf '::error title=Fatal error::"dune %s" exited with code %d\n' \
+        "$@" "$status" 1>&2
+      cat "$SETUPDUNEDIR/_build/log"
```

Comment:
I'm being silly and pedantic here but in theory the `_build` dir is also configurable...

However it's perfectly fine if `setup-dune` does not support this.

Reply:
Good point :smile: But, as you say, we should probably not target that level of configurability for an all-automagic action.